libsnark/gadgetlib1/gadgets/pairing/bw6_761_bls12_377/bls12_377_membership_check_gadgets.hpp

```diff
@@ -0,0 +1,98 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libsnark, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef LIBSNARK_GADGETLIB1_GADGETS_PAIRING_BW6_761_BLS12_377_BLS12_377_MEMBERSHIP_CHECK_GADGETS_HPP_
+#define LIBSNARK_GADGETLIB1_GADGETS_PAIRING_BW6_761_BLS12_377_BLS12_377_MEMBERSHIP_CHECK_GADGETS_HPP_
+
+#include "libsnark/gadgetlib1/gadgets/curves/weierstrass_g1_gadget.hpp"
+#include "libsnark/gadgetlib1/gadgets/curves/weierstrass_g2_gadget.hpp"
+#include "libsnark/gadgetlib1/gadgets/fields/fp12_2over3over2_gadgets.hpp"
+
+#include <libff/algebra/curves/bls12_377/bls12_377_pp.hpp>
+
+namespace libsnark
+{
+
+/// Curve equation (via the generic G1_checker_gadget) and subgroup membership
+/// check for BLS12-377 G1 variables.
+template<typename wppT>
+class bls12_377_G1_membership_check_gadget : gadget<libff::Fr<wppT>>
```

Comment:
`class bls12_377_G1_membership_check_gadget : gadget<libff::Fr<wppT>>` -> `class bls12_377_G1_membership_check_gadget : public gadget<libff::Fr<wppT>>` (unless private inheritance is strictly necessary here)

Reply:
updated to inherit publicly